packages/core/src/node-execution-context/base-execute-context.ts

```diff
@@ -122,9 +127,11 @@ export class BaseExecuteContext extends NodeExecutionContext {
 			parentCallbackManager,
 		});
 
-		// If a sub-workflow execution goes into the waiting state, then put the parent workflow execution also into the waiting state
+		// If a sub-workflow execution goes into the waiting state
 		if (result.waitTill) {
-			await this.putExecutionToWait(result.waitTill);
+			// then put the parent workflow execution also into the waiting state,
+			// but do not use the sub-workflow `waitTill` to avoid WaitTracker resuming the parent execution at the same time as the sub-workflow
+			await this.putExecutionToWait(new Date(WAIT_TIME_UNLIMITED));
```

Comment:
I remember this is what you mentioned back at the office. Maybe we should have a `putExecutionToWaitIndefinitely`

Reply:
I'd rather keep this explicit, without defaults. Another option: `putExecutionToWait({ indefinite: true })`